From e43876a9359c0299dbd1157c7173833d425af939 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Mon, 17 Aug 2020 01:00:29 +0800
Subject: [PATCH 3/3] video: fbdev: nvidia: Only match 0x0179 here

Geforce Go MX420M 32M uses ID 10de:0179 on Mac, which doesn't work with
nouveau driver.

Match it with nvidiafb, and then it can prevent offb and nouveau to
probe.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/video/fbdev/nvidia/nvidia.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/nvidia/nvidia.c b/drivers/video/fbdev/nvidia/nvidia.c
index fbeeed5afe35..74a7b9ce62b6 100644
--- a/drivers/video/fbdev/nvidia/nvidia.c
+++ b/drivers/video/fbdev/nvidia/nvidia.c
@@ -56,8 +56,7 @@
 #define MAX_CURS		32
 
 static const struct pci_device_id nvidiafb_pci_tbl[] = {
-	{PCI_VENDOR_ID_NVIDIA, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
-	 PCI_BASE_CLASS_DISPLAY << 16, 0xff0000, 0},
+	{ PCI_DEVICE(PCI_VENDOR_ID_NVIDIA, 0x0179) }, /* For GeForce Go MX420 32M, which is special on Mac */
 	{ 0, }
 };
 MODULE_DEVICE_TABLE(pci, nvidiafb_pci_tbl);
-- 
2.27.0

